### PR TITLE
Django ReST Framework

### PR DESCRIPTION
**Pull request by justquick**
- Dynamic serializers and viewsets for the registered models in the actstream.registry
- Default config for Action/Follow
- Views to match (most) streams in the action manager
- Ability to view your own actions
- Custom serializer, viewset and permission options in settings
- Related field evaluation using `rest-framework-generic-relations`

not ready for merge yet

#502 #286

### CODE REVIEW

File: runtests/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,9 @@
-Django>=2.2.17
+Django>=4.0
```
Comment (auvipy) [inline runtests/requirements.txt:1]: not even django 3.2 lts? or we will target to  4-4.2 versions to be minimal?

**Comment by mofe23** [comment]
If i try to install from this branch instead from pypi i get an ImportError.

Adding `actstream.drf` in `setup.py` resolves the error for me, see this [commit](https://github.com/mofe23/django-activity-stream/commit/376ed392e9530626f982d9479deb6f2376e3a0f6) on my fork:
``` python
      packages=['actstream',
                'actstream.drf',
                'actstream.migrations',
                'actstream.templatetags',
                'actstream.tests',
                ],
```

File: actstream/urls.py [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
 
+urlpatterns = []
 
-urlpatterns = [
+if USE_DRF:
+    from actstream.drf.urls import router
+    from testapp.views import SchemaExtender
+
+    urlpatterns += [
+        path('api/', include(SchemaExtender(router).urls)),
```
Comment (mofe23) [inline actstream/urls.py:13]: Remove dependency on SchemaeExtender from testapp.

**Review by mofe23 — CHANGES_REQUESTED** [review]
Im currently in the process of trying this PR, will add findings here if more occure

If i try to install from this branch instead of using pypi, i get an ImportError on startup.

Adding `actstream.drf` in `setup.py` resolves the error for me, see this [commit](https://github.com/mofe23/django-activity-stream/commit/376ed392e9530626f982d9479deb6f2376e3a0f6) on my fork:
``` python
      packages=['actstream',
                'actstream.drf',
                'actstream.migrations',
                'actstream.templatetags',
                'actstream.tests',
                ],
```

**Comment by auvipy** [comment]
is there any work remain from your side?

**Comment by justquick** [comment]
i need to see how dependent this needs to be on drf_spectacular.... it's a great feature but should be optional. id like to publish the output schema api explorer it generates somewhere but it shouldnt b required when running tests or anything else. those parts also need some work but can probably wait

**Comment by justquick** [comment]
im really tempted to merge this now but there are a couple things lacking. this really needs better tests and documentation. also we should push the openapi standard and provide a redoc site iamo. here's a run i did for the actstream drf api spec.

https://justquick.io/django-activity-stream/redoc.html?

**Comment by auvipy** [comment]
I checked the doc, what more you intended to complete?

**Comment by justquick** [comment]
we need to have a guide in the django-activity-stream documentation about how to use drf in actstream. just a page in the docs describing how to access it is enough. ill take a hack at something

i also want to strip the spectacular stuff out of the test app. just seems overkill and confusing. will add it in a later release

**Comment by justquick** [comment]
there are also changes to the `ACTSTREAM_SETTINGS` that add a `DRF` section which also needs documenting. hoping to finish this up soon

**Comment by justquick** [comment]
decided to rip off the bandaid and put this into main. please check it out and if there arent any issues then ill cut a 2.0 release

**Pull request merged**